src/ethereum/prague/fork.py

```diff
@@ -917,18 +915,23 @@ def process_transaction(
             for key in keys:
                 preaccessed_storage_keys.add((address, key))
 
-    message = prepare_message(
-        sender,
-        tx.to,
-        tx.value,
-        tx.data,
-        gas,
-        env,
-        preaccessed_addresses=frozenset(preaccessed_addresses),
-        preaccessed_storage_keys=frozenset(preaccessed_storage_keys),
-    )
-
-    output = process_message_call(message, env)
+    try:
+        message = prepare_message(
+            sender,
+            tx.to,
+            tx.value,
+            tx.data,
+            gas,
+            env,
+            preaccessed_addresses=frozenset(preaccessed_addresses),
+            preaccessed_storage_keys=frozenset(preaccessed_storage_keys),
+        )
+    except InvalidEof:
+        output = MessageCallOutput(
+            gas, U256(0), tuple(), set(), set(), InvalidEof(), b""
+        )
```

Comment:
I think we should preserve the original exception.

```suggestion
    except InvalidEof as error:
        output = MessageCallOutput(
            gas, U256(0), tuple(), set(), set(), error, b""
        )
```